cmake/profile.cmake

```diff
@@ -1,5 +1,5 @@
 set(CMAKE_REQUIRED_FLAGS "-fprofile-arcs -ftest-coverage")
-if (CMAKE_C_COMPILER_ID STREQUAL "GNU" AND NOT CMAKE_C_COMPILER_VERSION VERSION_LESS 11)
+if (CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 11)
```

Comment:
For GCC the reason is described in commit message bd813168467e4ac614439f56126cbce4134d493c, for Clang the reason why the version should be greater than 11 is not clear.
I propose to add a comment for both GCC and Clang and probably change condition to

`(compiler == GCC && version >= 11) or (compiler == Clang)`

Reply:
Ok, I've did some research) and the same change as in GCC 11 was introduced in llvm 12. https://github.com/llvm/llvm-project/commit/5809a32e7c2d79a9a463eb9c15cde994b42e3002
I've updated the PR to check for that and tested it with clang 12 and 18. I tried getting a clang 11 enviroment for my nixos, but it has been removed from repositories: https://github.com/NixOS/nixpkgs/pull/283591. If you insist n testing - I could get some old container with clang 11 and test it there, but rather wouldn't)

Reply:
Ok, I've did some research) and the same change as in GCC 11 was introduced in llvm 12. https://github.com/llvm/llvm-project/commit/5809a32e7c2d79a9a463eb9c15cde994b42e3002
I've updated the PR to check for that and tested it with clang 12 and 18. I tried getting a clang 11 enviroment for my nixos, but it has been removed from repositories: https://github.com/NixOS/nixpkgs/pull/283591. If you insist n testing - I could get some old container with clang 11 and test it there, but rather wouldn't)

Reply:
Thanks for the fix!

> Ok, I've did some research) and the same change as in GCC 11 was introduced in llvm 12. https://github.com/llvm/llvm-project/commit/5809a32e7c2d79a9a463eb9c15cde994b42e3002

I would add a link at least to the commit message.

Reply:
Done!